### PR TITLE
[240] Viewing provider users

### PR DESCRIPTION
**Pull request by darokel**
### Context

https://trello.com/c/NPwvxgB0/240-add-user-to-org-via-support-viewing-provider-users

### Changes proposed in this pull request

- Scopes the `/support/users` paths into `/providers/:provider_id/users/:user_id`
- Updates views and links
- Fixes tests
- Change links not implemented yet (in follow up ticket)
- Delete functionality not implemented yet (in sep ticket)

### Guidance to review

- Login as support and view users in the context of a provider
- Assert url is scoped to provider and view looks like prototype https://support-publish-prototype.herokuapp.com/organisations/82fd3a3b-6bfd-40d1-8343-0c19e34a847a/users

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Add user to org via support - viewing provider users](https://trello.com/c/NPwvxgB0/240-add-user-to-org-via-support-viewing-provider-users)

File: spec/features/support/providers/viewing_a_user_spec.rb [hunk 1]
```diff
@@ -13,15 +13,16 @@
...
+    then_i_am_directed_to_the_provider_user_show_page
   end
 
   def when_i_visit_the_support_provider_show_page
     support_provider_show_page.load(id: @user.providers.first.id)
   end
 
   def and_there_is_a_user
-    @user = create(:user, :with_provider)
+    @user = create(:user, :with_provider, last_login_date_utc: Time.zone.yesterday)
```
Comment (defong) [inline spec/features/support/providers/viewing_a_user_spec.rb:24]: This is hiding a problem, as not everyone will have a `last_login_date_utc`value

File: app/views/support/providers/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+          component.row do |row|
+            row.key { "Organisations" }
+            row.value(text: sanitize(@user.providers.map(&:provider_name).join(tag.br)), html_attributes: { id: "organisations" })
+            row.action
+          end
+
+          component.row do |row|
+            row.key { "Date last signed in" }
+            row.value(text: @user.last_login_date_utc.strftime("%d %B %Y at %I:%M%p"), html_attributes: { id: "date_last_signed_in" })
```
Comment (defong) [inline app/views/support/providers/users/show.html.erb:44]:
```suggestion
            row.value(text: @user.last_login_date_utc&.strftime("%d %B %Y at %I:%M%p"), html_attributes: { id: "date_last_signed_in" })
```

Comment (defong) [reply app/views/support/providers/users/show.html.erb:44]: https://publish-teacher-training-pr-2838.london.cloudapps.digital/support/providers/18143/users/11918

File: spec/features/support/providers/viewing_a_user_spec.rb (hunk 1 above)
Comment (darokel) [reply spec/features/support/providers/viewing_a_user_spec.rb]: yup agreed, have split out the scenarios

**Review by defong — APPROVED** [review]

File: app/views/support/providers/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+          component.row do |row|
+            row.key { "Email address" }
+            row.value(text: @user.email, html_attributes: { id: "email" })
+            row.action
+          end
+
+          component.row do |row|
+            row.key { "Organisations" }
+            row.value(text: sanitize(@user.providers.in_current_cycle.map(&:provider_name).join(tag.br)), html_attributes: { id: "organisations" })
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/views/support/providers/users/show.html.erb]: yup copy pasta from another view 😩

**Pull request merged**